### PR TITLE
ammonite-repl: new ports 2.2.0

### PR DESCRIPTION
**Pull request by josh-seidel-db**
* add ammonite-repl2.12 version 2.2.0 for scala2.12
* add ammonite-repl2.13 version 2.2.0 for scala2.13
* add ammonite-repl_select for version selection based on scala

see: https://trac.macports.org/ticket/61394

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by josh-seidel-db** [comment]
Hello, just checking in to see if someone would review the pull request and add in the port to the macports-ports repo? I will continue to maintain it as part of my work and having it as a port is very useful. This is a great tool. Thank you! Cheers, Josh

**Comment by reneeotten** [comment]
hi @josh-seidel-db apologies for the fact that nobody has found the time or interest to review your PRs... I must admit that I have no idea about the software nor Java, but perhaps can help you to get the Portfiles in such a shape that they can be merged - that is, if you are still interested to work on this!

If so, let's pick one of the to PRs you submitted and give it a go. I guess my first question would be do we actually need ports for different versions of can we just pick the latest-and-greatest one and stick to that. Along these lines, if multiple versions are required it seems they could perhaps better be implemented as subports given the high similarity and repetition in the different portfiles.

File: java/ammonite-repl2.12/Portfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    A Modernized Scala REPL. With syntax highlighting, multi-line editing, the ability \
+    to load maven artifacts directly in the REPL, and many other quality-of-life \
+    improvements missing in the default Scala REPL.
+homepage            https://ammonite.io/
+platforms           darwin
+supported_archs     noarch
+license             MIT
+use_xcode           no
+
+master_sites        https://github.com/lihaoyi/Ammonite/releases/download/${version}/
```
Comment (reneeotten) [inline java/ammonite-repl2.12/Portfile:22]: given that this comes from GitHub, you should use the github PG

File: java/ammonite-repl2.12/Portfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+description         Engine for large-scale data processing
+long_description \
+    A Modernized Scala REPL. With syntax highlighting, multi-line editing, the ability \
+    to load maven artifacts directly in the REPL, and many other quality-of-life \
+    improvements missing in the default Scala REPL.
+homepage            https://ammonite.io/
+platforms           darwin
+supported_archs     noarch
+license             MIT
+use_xcode           no
```
Comment (reneeotten) [inline java/ammonite-repl2.12/Portfile:20]: this is the default and therefore not needed

File: java/ammonite-repl2.12/Portfile [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+master_sites        https://github.com/lihaoyi/Ammonite/releases/download/${version}/
+distname            2.12-${version}
+distfiles           ${distname}
+checksums           md5    bc96c0c355473e449d14b3e8c080f038\
+                    sha1   377b4a46dcf363534d85ed2be929ce592c0182f7\
+                    rmd160 67e33d7415a171f87896b64730a62879a6c818e3\
+                    sha256 f717b8193e58a550ad0cf699be2f4f670156ff070d860f9009d6c9e139769459\
+                    size   44937814
```
Comment (reneeotten) [inline java/ammonite-repl2.12/Portfile:29]: no need for `md5` and `sha1`

File: java/ammonite-repl2.12/Portfile [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+# Require java version
+java.version        8+
+set java.base.version 8
+# JDK port to install if required java not found
+java.fallback       openjdk8
+
+depends_run         port:scala2.12
+
+if {$subport == $name} {
```
Comment (reneeotten) [inline java/ammonite-repl2.12/Portfile:41]: it seems to me that there is no need for this `if`-block

File: java/ammonite-repl2.12/Portfile [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    depends_lib          port:ammonite-repl_select
```
Comment (reneeotten) [inline java/ammonite-repl2.12/Portfile:42]: this should be `depends_run` as it is not needed at build time

File: java/ammonite-repl2.12/Portfile [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    select.group         ammonite-repl
+    select.file          ${filespath}/${name}
+
+    extract {
+    	file copy ${prefix}/var/macports/distfiles/${name} ${worksrcpath}
+    }
+
+    set ammonite.cmd    amm
+
+    build.cmd           ${distfiles}
```
Comment (reneeotten) [inline java/ammonite-repl2.12/Portfile:52]: what is this supposed to do?

File: java/ammonite-repl2.12/Portfile [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+depends_run         port:scala2.12
+
+if {$subport == $name} {
+    depends_lib          port:ammonite-repl_select
+    select.group         ammonite-repl
+    select.file          ${filespath}/${name}
+
+    extract {
+    	file copy ${prefix}/var/macports/distfiles/${name} ${worksrcpath}
+    }
```
Comment (reneeotten) [inline java/ammonite-repl2.12/Portfile:48]: why do you need to override the `extract` phase?

**Comment by reneeotten** [comment]
Closing this due to inactivity - feel free to reopen once you are ready and we can try to get this merged!

**Pull request closed without merging**